azure-iot-device/tests/common/pipeline/pipeline_stage_test.py

```diff
@@ -14,29 +14,202 @@
     assert_callback_failed,
     get_arg_count,
     add_mock_method_waiter,
-    StageTestBase,
     StageRunOpTestBase,
     StageHandlePipelineEventTestBase,
 )
 from azure.iot.device.common.pipeline.pipeline_stages_base import PipelineStage, PipelineRootStage
-from tests.common.pipeline.pipeline_event_test import add_instantiation_test
 from azure.iot.device.common.pipeline import pipeline_thread, pipeline_exceptions
 from azure.iot.device.common import handle_exceptions
-from .fixtures import FakeOperation
-from .helpers import assert_callback_succeeded
+from .fixtures import ArbitraryOperation
 
 logging.basicConfig(level=logging.DEBUG)
 
 
 def add_base_pipeline_stage_tests(
+    test_module,
+    stage_class_under_test,
+    stage_test_config_class,
+    extended_stage_instantiation_test_class=None,
+):
+    class StageTestConfig(stage_test_config_class):
+        @pytest.fixture
+        def stage(self, mocker, cls_type, init_kwargs):
+            stage = cls_type(**init_kwargs)
+            stage.next = mocker.MagicMock()
+            stage.previous = mocker.MagicMock()
+            mocker.spy(stage, "send_op_down")
+            mocker.spy(stage, "send_event_up")
+            return stage
+
+    #######################
+    # INSTANTIATION TESTS #
+    #######################
+
+    @pytest.mark.describe("{} -- Instantiation".format(stage_class_under_test.__name__))
+    class StageBaseInstantiationTests(StageTestConfig):
+        @pytest.mark.it("Initializes 'name' attribute as the classname")
+        def test_name(self, cls_type, init_kwargs):
+            stage = cls_type(**init_kwargs)
```

Comment:
>stage [](start = 12, length = 5)

nit: why not use the stage fixture?  Here and everywhere cls_type is called with **init_kwargs and nothing else.

Reply:
As mentioned elsewhere, I feel that the unit under test (instantiation) should occur within the test itself, rather than a fixture.

---
In reply to: [354869536](https://github.com/Azure/azure-iot-sdk-python/pull/409#discussion_r354869536) [](ancestors = 354869536)